torch/csrc/api/include/torch/data/transforms/tensor.h

```diff
@@ -50,6 +50,23 @@ class TensorLambda : public TensorTransform<Target> {
  private:
   FunctionType function_;
 };
+
+/// Normalizes input tensors by subtracting the supplied mean and dividing by
+/// the given standard deviation.
+template <typename Target = Tensor>
+struct Normalize : public TensorTransform<Target> {
+
+  /// Constructs a `Normalize` transform. The mean and standard deviation can be
+  //anything that is broadcastable over the input tensors (like single scalars).
+  Normalize(ArrayRef<float> mean, ArrayRef<float> stddev)
+      : mean(torch::tensor(mean)), stddev(torch::tensor(stddev)) {}
+
+  torch::Tensor operator()(Tensor input) {
+    return input.sub(mean).div(stddev);
```

Comment:
Question: don't you need to unsqueezr twice if your mean/std is not a scalar, because the batch dimension in the input (4d) is 1?

Reply:
Correct me if I'm wrong but from https://github.com/pytorch/vision/blob/master/torchvision/transforms/functional.py I figured the user is responsible for passing in the right thing if not a scalar?

Reply:
The link you pointed has the unsqueeze.
One thing to keep in mind is that torchvision normalize expect one value per channel. Maybe you plan on supporting arbitrary values (which change spatially) here as well?

Reply:
Hmm so if you pass a two-element tensor as the mean/stddev wouldn't that broadcast correctly over the channels of a two-channel image? At least that's what I test in the tests

Reply:
Yes, it will.

I think it all boils down to how we represent the images at this point: are they `CHW` (the torch way) or `HWC`?
The current implementation works for `HWC`.